src/source/CurlApiCallbacks.c

```diff
@@ -1896,25 +2025,83 @@ STATUS tagResourceCachingCurl(UINT64 customData, PCHAR streamArn, UINT32 tagCoun
                               PServiceCallContext pServiceCallContext)
 {
     ENTERS();
-    UNUSED_PARAM(streamArn);
-    UNUSED_PARAM(tagCount);
-    UNUSED_PARAM(tags);
-
     STATUS retStatus = STATUS_SUCCESS;
     PCurlApiCallbacks pCurlApiCallbacks = (PCurlApiCallbacks) customData;
     PCallbacksProvider pCallbacksProvider = NULL;
     STREAM_HANDLE streamHandle;
+    BOOL endpointsLocked = FALSE, emulateApiCall = TRUE;
+    UINT64 curTime, value;
+    PEndpointTracker pEndpointTracker = NULL;
 
     CHK(pCurlApiCallbacks != NULL && pCurlApiCallbacks->pCallbacksProvider != NULL && pServiceCallContext != NULL, STATUS_INVALID_ARG);
     pCallbacksProvider = pCurlApiCallbacks->pCallbacksProvider;
 
     streamHandle = (STREAM_HANDLE) pServiceCallContext->customData;
-    retStatus = tagResourceResultEvent(streamHandle, SERVICE_CALL_RESULT_OK);
 
+    // We check whether we have already made the call by checking
+    // for the presence of the endpoint in the cache.
+    switch(pCurlApiCallbacks->cacheType) {
+        case API_CALL_CACHE_TYPE_NONE:
+            emulateApiCall = FALSE;
+            break;
+
+        case API_CALL_CACHE_TYPE_ENDPOINT_ONLY:
+            emulateApiCall = TRUE;
+            break;
+
+        case API_CALL_CACHE_TYPE_ALL:
+            pCallbacksProvider->clientCallbacks.lockMutexFn(pCallbacksProvider->clientCallbacks.customData,

```

Comment:
Do we need all these? tagResourceResultEvent just take a stream handle.

Reply:
Not sure what you mean. This block checks what caching level needs to be done and whether we need to call the main function to actually perform the call